homeassistant/components/schluter/__init__.py

```diff
@@ -32,44 +43,78 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+def schluter_auth_update(hass_domain: dict[str,any]) -> bool:
+    ''' 
+    schluter_auth_update() updates session_id and expires inside the hass.data[DOMAIN].
+    returns None on error and session id on success
+
+    Note: we are reimplementing the code from Schluter's library Authenticator, because
+    that code dumps error message upon token expiration and this pollutes the logs. '''
+
+    api = hass_domain[DATA_SCHLUTER_API];
+    session = hass_domain[DATA_SCHLUTER_SESSION];
+    expires = hass_domain[DATA_SCHLUTER_EXPIRES];
+
+    if expires > datetime.utcnow():
+        return session;
+
+    _LOGGER.debug("schluter: reacquiring new session ID.");
+
+    try:
+        ## 
+        ## As of 2022, one http session to Schluter cloud can last for months.
+        ## Thus we could reuse it between the session id updates,
+        ## but it seems to be safer to reestablish it periodically. 
+        ##
+        ## HACK: This is probably not a good idea because _http_session is not exposed by the API.
+        api._http_session.close();
+
+        response = api.get_session(hass_domain[DATA_SCHLUTER_USER],hass_domain[DATA_SCHLUTER_PASS]);
```

Comment:
This should be handled by the client library and not here in the integration. The library should raise library specific exceptions when there are problems if the library user need to take action to resolve the issue.